# Table of Contents
 <p><div class="lev1 toc-item"><a href="#Project-1:-Deaths-by-tuberculosis" data-toc-modified-id="Project-1:-Deaths-by-tuberculosis-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Project 1: Deaths by tuberculosis</a></div><div class="lev2 toc-item"><a href="#Env" data-toc-modified-id="Env-11"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Env</a></div><div class="lev2 toc-item"><a href="#The-data" data-toc-modified-id="The-data-12"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>The data</a></div><div class="lev2 toc-item"><a href="#The-range-of-the-problem" data-toc-modified-id="The-range-of-the-problem-13"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>The range of the problem</a></div><div class="lev2 toc-item"><a href="#The-most-affected" data-toc-modified-id="The-most-affected-14"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>The most affected</a></div><div class="lev2 toc-item"><a href="#Conclusions" data-toc-modified-id="Conclusions-15"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>Conclusions</a></div>

# Project 1: Deaths by tuberculosis

by Michel Wermelinger, 14 July 2015, with minor edits on 5 April 2016<br>

This is the project notebook for Week 1 of The Open University's [_Learn to code for Data Analysis_](http://futurelearn.com/courses/learn-to-code) course.

In 2000, the United Nations set eight Millenium Development Goals (MDGs) to reduce poverty and diseases, improve gender equality and environmental sustainability, etc. Each goal is quantified and time-bound, to be achieved by the end of 2015. Goal 6 is to have halted and started reversing the spread of HIV, malaria and tuberculosis (TB).
TB doesn't make headlines like Ebola, SARS (severe acute respiratory syndrome) and other epidemics, but is far deadlier. For more information, see the World Health Organisation (WHO) page <http://www.who.int/gho/tb/en/>.

Given the population and number of deaths due to TB in some countries during one year, the following questions will be answered: 

- What is the total, maximum, minimum and average number of deaths in that year?
- Which countries have the most and the least deaths?
- What is the death rate (deaths per 100,000 inhabitants) for each country?
- Which countries have the lowest and highest death rate?

The death rate allows for a better comparison of countries with widely different population sizes.

## Env

In [1]:
# Print platform info of Python exec env.
import sys
sys.version

'3.6.1 |Anaconda custom (64-bit)| (default, May 11 2017, 13:25:24) [MSC v.1900 64 bit (AMD64)]'

In [2]:
import warnings
warnings.simplefilter('ignore', FutureWarning)

from pandas import *
show_versions()


INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.2
pytest: 3.1.2
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.1
xarray: None
IPython: 6.1.0
sphinx: 1.6.2
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.2.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.8.0
bs4: 4.6.0
html5lib: 0.999
sqlalchemy: 1.1.11
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: 0.4.0


## The data

The data consists of total population and total number of deaths due to TB (excluding HIV) in 2013 in each of the BRICS (Brazil, Russia, India, China, South Africa) and Portuguese-speaking countries. 

The data was taken in July 2015 from <http://apps.who.int/gho/data/node.main.POP107?lang=en> (population) and <http://apps.who.int/gho/data/node.main.593?lang=en> (deaths). The uncertainty bounds of the number of deaths were ignored.

The data was collected into an Excel file which should be in the same folder as this notebook.

In [3]:
data = read_excel('WHO POP TB some.xls')
data.head()

Country  Population (1000s)  TB deaths
0             Angola               21472       6900
1             Brazil              200362       4400
2              China             1393337      41000
3  Equatorial Guinea                 757         67
4      Guinea-Bissau                1704       1200

In [4]:
data.tail()

Country  Population (1000s)  TB deaths
7                Portugal               10608        140
8      Russian Federation              142834      17000
9   Sao Tome and Principe                 193         18
10           South Africa               52776      25000
11            Timor-Leste                1133        990

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 3 columns):
Country               12 non-null object
Population (1000s)    12 non-null int64
TB deaths             12 non-null int64
dtypes: int64(2), object(1)
memory usage: 368.0+ bytes


In [6]:
data.describe()

Population (1000s)      TB deaths
count        1.200000e+01      12.000000
mean         2.585958e+05   29559.583333
std          5.019379e+05   67489.518992
min          1.930000e+02      18.000000
25%          1.561250e+03     777.500000
50%          2.365300e+04    5650.000000
75%          1.572160e+05   19750.000000
max          1.393337e+06  240000.000000

## The range of the problem

The column of interest is the last one.

In [7]:
tbColumn = data['TB deaths']

The total number of deaths in 2013 is:

In [8]:
tbColumn.sum()

354715

The largest and smallest number of deaths in a single country are:

In [9]:
tbColumn.max()

240000

In [10]:
tbColumn.min()

18

From less than 20 to almost a quarter of a million deaths is a huge range. The average number of deaths, over all countries in the data, can give a better idea of the seriousness of the problem in each country.
The average can be computed as the mean or the median. Given the wide range of deaths, the median is probably a more sensible average measure.

In [11]:
tbColumn.mean()

29559.583333333332

In [12]:
tbColumn.median()

5650.0

The median is far lower than the mean. This indicates that some of the countries had a very high number of TB deaths in 2013, pushing the value of the mean up.

## The most affected

To see the most affected countries, the table is sorted in ascending order by the last column, which puts those countries in the last rows.

In [13]:
data.sort_values('TB deaths').head()

Country  Population (1000s)  TB deaths
9   Sao Tome and Principe                 193         18
3       Equatorial Guinea                 757         67
7                Portugal               10608        140
11            Timor-Leste                1133        990
4           Guinea-Bissau                1704       1200

The table raises the possibility that a large number of deaths may be partly due to a large population. To compare the countries on an equal footing, the death rate per 100,000 inhabitants is computed.

In [14]:
populationColumn = data['Population (1000s)']
data['TB deaths (per 100,000)'] = tbColumn * 100 / populationColumn
data.head()

Country  Population (1000s)  TB deaths  TB deaths (per 100,000)
0             Angola               21472       6900                32.134873
1             Brazil              200362       4400                 2.196025
2              China             1393337      41000                 2.942576
3  Equatorial Guinea                 757         67                 8.850727
4      Guinea-Bissau                1704       1200                70.422535

## Conclusions

The BRICS and Portuguese-speaking countries had a total of about 350 thousand deaths due to TB in 2013. The median shows that half of these coutries had fewer than 5,650 deaths. The much higher mean (over 29,000) indicates that some countries had a very high number. The least affected were Sao Tome and Principe and Equatorial Guinea, with 18 and 67 deaths respectively, and the most affected were China and India with 41 thousand and 240 thousand deaths in a single year. However, taking the population size into account, the least affected were Portugal and Brazil with less than 2.2 deaths per 100 thousand inhabitants, and the most affected were Guinea-Bissau and East Timor with over 70 deaths per 100,000 inhabitants.

One should not forget that most values are estimates, and that the chosen countries are a small sample of all the world's countries. Nevertheless, they convey the message that TB is still a major cause of fatalities, and that there is a huge disparity between countries, with several ones being highly affected.